app/controllers/api/v1/books_controller.rb

```
@@ -4,14 +4,23 @@ module Api
   module V1
     class BooksController < ApplicationController
       include Wor::Paginate
-      before_action :authenticate_user!
+      before_action :authenticate_user!, only: %i[index show]
       def index
         render_paginated Book
       end
 
       def show
         render Book.find(params[:id])
       end
+
+      def find_book
+        if request.headers['isbn']
+          book_request = BookLibraryApi.new(request.headers['isbn'])
+          render json: book_request.book
+        else
+          render json: { status: 'error', code: 400, message: 'Params missing.' }, status: :bad_request
```

Remember, It's better when you fail early.

Think like,

```
return render json: { status: 'error', code: 400, message: 'Params missing.' }, status: :bad_request unless request.headers['isbn']
```